pkg/controller/master/storagenetwork/storage_network.go

```diff
@@ -0,0 +1,310 @@
+package storagenetwork
+
+import (
+	"context"
+	"reflect"
+	"strconv"
+	"time"
+
+	longhornv1 "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta1"
+	//ctlcorev1 "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	"github.com/sirupsen/logrus"
+	//corev1 "k8s.io/api/core/v1"
+	//"github.com/rancher/wrangler/pkg/condition"
+	v1 "github.com/rancher/wrangler/pkg/generated/controllers/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/config"
+	ctlharvesterv1 "github.com/harvester/harvester/pkg/generated/controllers/harvesterhci.io/v1beta1"
+	//ctlkubevirtv1 "github.com/harvester/harvester/pkg/generated/controllers/kubevirt.io/v1"
+	ctllonghornv1 "github.com/harvester/harvester/pkg/generated/controllers/longhorn.io/v1beta1"
+	ctlmonitoringv1 "github.com/harvester/harvester/pkg/generated/controllers/monitoring.coreos.com/v1"
+	"github.com/harvester/harvester/pkg/settings"
+	"github.com/harvester/harvester/pkg/util"
+	ctlmgmtv3 "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+)
+
+const (
+	ControllerName                  = "harvester-backup-target-controller"
+	StorageNetworkAnnotation        = "storage-network.settings.harvesterhci.io"
+	ReplicaStorageNetworkAnnotation = StorageNetworkAnnotation + "/replica"
+	PausedStorageNetworkAnnotation  = StorageNetworkAnnotation + "/paused"
+
+	longhornStorageNetworkName = "storage-network"
+
+	// Rancher monitoring
+	CattleMonitoringSystemNamespace = "cattle-monitoring-system"
```

Comment:
better to put those into `pkg/util/constants.go`

Reply:
i'm thinking if we don't need it in other place, do we need to put it in constants.go?
I will check it
thanks a lot

Reply:
All `namespace`s should be put into constants, they will be reused now and then.

e.g. `cattle-monitoring-system` has already been locally defined in following files, it's not good.
```
pkg/controller/master/rancher/embedded_rancher.go:		"cattle-monitoring-system",
pkg/controller/master/supportbundle/manager.go:		"cattle-monitoring-system",
```

Those are better to be in constants.
```
	RancherMonitoringPrometheus     = "rancher-monitoring-prometheus"
	FleetLocalNamespace             = "fleet-local"
	RancherMonitoring               = "rancher-monitoring"
	RancherMonitoringGrafana        = "rancher-monitoring-grafana"
```